src/client/components/builder/logic/ConditionalResult.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useState, useEffect } from 'react'
+import update from 'immutability-helper'
+import { BiGitBranch, BiPlusCircle, BiTrash } from 'react-icons/bi'
+import {
+  Divider,
+  Button,
+  IconButton,
+  VStack,
+  HStack,
+  Box,
+  Text,
+  Input,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+} from '@chakra-ui/react'
+
+import { useCheckerContext } from '../../../contexts'
+import { createBuilderField, OperationFieldComponent } from '../BuilderField'
+import { BuilderActionEnum, ConfigArrayEnum } from '../../../../util/enums'
+
+interface Condition {
+  type: 'AND' | 'OR'
+  expression: string
+}
+
+interface IfelseState {
+  ifExpr: string
+  conditions: Condition[]
+  elseExpr: string
+  thenExpr: string
+}
+
+const EMPTY_STATE: IfelseState = {
+  ifExpr: '',
+  conditions: [],
+  elseExpr: '',
+  thenExpr: '',
+}
+
+const fromExpression = (expression: string): IfelseState => {
```

Comment:
I guess a more robust way will be to use `mathjs` to parse this and then traverse the expression tree to build the string. Will probably get back to that in the future.